lib/bundler_api/versions_file.rb

```diff
@@ -0,0 +1,66 @@
+require 'bundler_api'
+
+class BundlerApi::VersionsFile
+  PATH = "versions.list"
+  def initialize(connection)
+    @conn = connection
+  end
+
+  def update(file = nil)
+    file ||= PATH
+
+    content = Time.now.to_i.to_s
+    content += "\n---\n"
+    content += gems_with_versions
+
+    File.open(file, 'w') do |io|
+      io.write content
+    end
+  end
+
+  def with_new_gems
+    if gems_with_versions(created_at).empty?
+      content
+    else
+      content + "\n" + gems_with_versions(created_at)
+    end
+  end
+
+  private
+    def content
+      File.open(PATH).read
+    end
+
+    def created_at
+      DateTime.strptime(File.open(PATH).readline, "%s")
+    end
+
+    def gems_with_versions(newer_than = nil)
+      if newer_than
+        dataset = @conn[<<-SQL, newer_than]
+            SELECT
+              r.name, string_agg(
+                        concat_ws('-', v.number, nullif(v.platform,'ruby')), ','
+                        ORDER BY number ASC
+                      )
+            FROM rubygems AS r, versions AS v
+            WHERE v.rubygem_id = r.id AND
+                  v.indexed is true AND
+                  v.created_at > ?
+            GROUP BY r.name
```

Comment:
That was fast! Not sure if I got you. This thing is actually only getting the new gems. The result is concatenated with the static file on line 25. Note the created_at. So the first lines on the response won't change. Not sure why the group by is getting in the way?